canopy/canopyjs/src/transactions/signing.ts

```diff
@@ -0,0 +1,163 @@
+/**
+ * Copyright 2019 Cargill Incorporated
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import crypto from 'crypto';
+import secp256k1 from 'secp256k1';
+
+function toHex(buffer: Uint8Array): string {
+  return Array.from(buffer)
+    .map(b => b.toString(16).padStart(2, '0'))
+    .join('');
+}
+
+interface IContext {
+  sign: (message: Uint8Array, privateKey: PrivateKey) => string;
+  getPublicKey: (privateKey: PrivateKey) => PublicKey;
+}
+
+interface IPrivateKey {
```

Comment:
I'd recommend splitting the interfaces and implementations out into separate files, see the rust sdk for example: [Rust SDK](https://github.com/hyperledger/sawtooth-sdk-rust/tree/master/src/signing).

Traits would be interfaces and structs would be classes.